feat: drop support for php < 8.1

Chris53897 opened the pull request:
drop support for symfony < 5.4
add "ext-simplexml" as required
add strong types

As i was working in https://github.com/moneyphp/iso-currencies/pull/14
i made some more changes for a new major version. There are BC breaks, and i am not sure about the changes in the interface.

Chris8934 committed:
feat: drop support for php < 8.1
drop support for symfony < 5.4
add "ext-simplexml" as required
add strong types

frederikbosch commented on composer.json:
```diff
@@ -13,8 +13,9 @@
     "require": {
-        "php": ">=5.5",
-        "symfony/yaml": "^3.2 | ^4.0 | ^5.0 | ^6.0"
+        "php": ">=8.1",
```
~8.1.0 || ~8.2.0

Chris53897 replied:
You mean ~8.1.0 || ~8.2.0 || ~8.3.0 ?

frederikbosch commented on src/Country.php:
```diff
@@ -1,85 +1,37 @@
-     * @param int $minorUnit
-     */
-    public function __construct($entity, $currency, $alphabeticCode, $numericCode, $minorUnit)
+    public function __construct(private readonly string $entity,
```
Add new line

frederikbosch commented on src/Country.php:
```diff
@@ -1,85 +1,37 @@
+                                private readonly string $currency,
+                                private readonly string $alphabeticCode,
+                                private readonly int    $numericCode,
+                                private readonly int    $minorUnit)
```
Add new line, identation

Chris53897 replied:
i changed identation betwen type and variable name. I am not sure if you that is what you mean.

I am not sure where exact you want me to add the new line

frederikbosch reviewed (commented):
Great work!

frederikbosch commented:
Let's get this is in, with the minor changes I suggested.

Chris8934 committed:
feat: drop support for php < 8.1
drop support for symfony < 5.4
add "ext-simplexml" as required
add strong types

frederikbosch commented:
The two new lines and indentation are still left. If you correct them I will press the button.

Chris8934 committed:
feat: drop support for php < 8.1
drop support for symfony < 5.4
add "ext-simplexml" as required
add strong types

Chris53897 commented:
I am not sure how exactly you want it.
I did change it now, but maybe wrong
